examples/piecestore/rpc/client/api/api.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+  "io"
+  "log"
+
+  "golang.org/x/net/context"
+
+  "google.golang.org/grpc"
+
+  "github.com/zeebo/errs"
+
+  pb "storj.io/storj/examples/piecestore/rpc/protobuf"
+)
+
+var ServerError = errs.Class("serverError")
+
+// Struct containing Shard information from ShardMetaRequest
+type ShardMeta struct {
+  Hash       string
+  Size       int64
+  Expiration int64
+}
+
+// Request info about a shard by Shard Hash
+func ShardMetaRequest(conn *grpc.ClientConn, hash string) (*ShardMeta, error) {
+  c := pb.NewRouteGuideClient(conn)
+
+  reply, err := c.Shard(context.Background(), &pb.ShardHash{Hash: hash})
+  if err != nil {
+    return nil, err
+  }
+
+  return &ShardMeta{Hash: reply.Hash, Size: reply.Size, Expiration: reply.Expiration}, nil
+}
+
+// Upload Shard to Server
+func StoreShardRequest(conn *grpc.ClientConn, hash string, data io.Reader, dataOffset int64, length int64, ttl int64, storeOffset int64) (error) {
+  c := pb.NewRouteGuideClient(conn)
+
+  stream, err := c.Store(context.Background())
+
+  buffer := make([]byte, 4096)
+  for {
+    // Read data from read stream into buffer
+    n, err := data.Read(buffer)
+    if err == io.EOF {
```

Comment:
What if an error occur that is different than EOF? Won't it go in an infinite cycle?